### PR TITLE
[qdk-python] Update Quantinuum target names

### PR DESCRIPTION
**Pull request by warren-jones**
The Quantinuum target names are changing:

Old name | New name
-- | --
quantinuum.hqs-lt-s1 | quantinuum.qpu.h1-1
quantinuum.hqs-lt-s1-apival | quantinuum.sim.h1-1sc
quantinuum.hqs-lt-s2 | quantinuum.qpu.h1-2
quantinuum.hqs-lt-s2-apival | quantinuum.sim.h1-2sc
quantinuum.hqs-lt-s1-sim | quantinuum.sim.h1-1e
quantinuum.hqs-lt-s2-sim | quantinuum.sim.h1-2e
quantinuum.hqs-lt | quantinuum.qpu.h1

Adding new target names where old names appear, 10 occurrences in operational code, 4 in tests.  Leaving old names in place until they are deprecated.

The deprecation date for the old target names is TBD.

### CODE REVIEW

**Review by shenyingjun — APPROVED** [review]

**Comment by warren-jones** [comment]
@vxfield ,
All of the modified tests that would access the service were already marked pytest.mark.live_test.  The cost estimate tests don't need the recordings, and the others are live-only, so are you OK with not updating the recordings on this PR?

File: azure-quantum/azure/quantum/target/quantinuum.py [hunk 1]
```diff
@@ -17,12 +17,15 @@ class Quantinuum(Target):
         "quantinuum.hqs-lt-s1",
         "quantinuum.hqs-lt-s1-apival",
         "quantinuum.hqs-lt-s1-sim",
+        "quantinuum.qpu.h1-1",
```
Comment (xinyi-joffre) [inline azure-quantum/azure/quantum/target/quantinuum.py:20]: It seems like only h1-1 targets are here, do you know why all the h1-2 versionso of the targets seem to be missing from this list?

Comment (xinyi-joffre) [reply azure-quantum/azure/quantum/target/quantinuum.py:20]: @ricardo-espinoza, do you have any context on why only h1-1 targets are in this target_name set?

Comment (warren-jones) [reply azure-quantum/azure/quantum/target/quantinuum.py:20]: All the testing was done with what you see here -- only hqs-lt-s1 names were present before -- and cut-off for September QDK release is tomorrow morning.  Could we create a bug to add h1-2 names in a later release?

Comment (warren-jones) [reply azure-quantum/azure/quantum/target/quantinuum.py:20]: @xinyi-joffre , @ricardo-espinoza ,
I added the h1-2 target names in azure-quantum/azure/quantum/target/quantinuum.py and in the unit tests.  Running the tests again.

Comment (ricardo-espinoza) [reply azure-quantum/azure/quantum/target/quantinuum.py:23]: It seems that the related PR #371 hasn't been merged. If that had been merged earlier, then Warren's approach of just adding the new names where the old names were would have been enough.

Comment (ricardo-espinoza) [reply azure-quantum/azure/quantum/target/quantinuum.py:23]: I would recommend we exclude the -2 targets until the related PR is merged. Other than that, LGTM.

Comment (warren-jones) [reply azure-quantum/azure/quantum/target/quantinuum.py:20]: I removed the h1-2 target names.

**Pull request merged**